### PR TITLE
[CP-2000] [Messages] Search results seems to be in random order.

### PR DESCRIPTION
**Pull request by patryk-sierzega**
Jira: [CP-2000]
**Description**

- changed filtering of receivers and contacts - now these are filtered by firstName, lastName, phoneNumber (one or two) and fullName (which is firstName + lastName)
- changed name of flatlist to contacts
- changed selector getFlatList to getContacts
- [ ] getState function in async thunk actions is correctly typed
- [ ] redux selectors are used in components / prop drilling is reduce

[CP-2000]: https://appnroll.atlassian.net/browse/CP-2000?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/contacts/contacts.container.tsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ const mapStateToProps = (state: RootModel & ReduxRootState) => {
     allItemsSelected: state.contacts.selectedItems.allItemsSelected,
     resultState: contacts.resultState,
     contactList: contactsList,
-    flatList: flatListSelector(state),
+    contacts: contactsSelector(state),
```
Comment (dkarski) [inline packages/app/src/contacts/contacts.container.tsx:56]: 👍

File: packages/app/src/utils/sort-by-last-name.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+export const sortByLastNameAscending = <T extends { lastName?: string }>(
```
Comment (dkarski) [inline packages/app/src/utils/sort-by-last-name.ts:6]: The function name seems to be isn't equal to a file name. Is it intentional?

File: packages/app/src/contacts/components/contacts/contacts.component.tsx [hunk 3]
```diff
@@ -51,12 +51,12 @@ import ImportContactsFlow, {
 } from "App/contacts/components/import-contacts-flow/import-contacts-flow.component"
 import { Contact, NewContact } from "App/contacts/reducers/contacts.interface"
 import { isError } from "App/__deprecated__/common/helpers/is-error.helpers"
-import { contactsFilter } from "App/contacts/helpers/contacts-filter/contacts-filter.helper"
 import { ExportContactFailedModal } from "../export-contact-failed-modal/export-contact-failed-modal.component"
 import { applyValidationRulesToImportedContacts } from "App/contacts/helpers/apply-validation-rules-to-imported-contacts/apply-validation-rules-to-imported-contacts"
 import { ExportContactsResult } from "App/contacts/constants"
 import DeleteContactsPopup from "./delete-contacts-popup/delete-contacts-popup.component"
 import { differenceWith, isEqual } from "lodash"
+import { filterContacts } from "App/messages/helpers/messages.helpers"
```
Comment (dkarski) [inline packages/app/src/contacts/components/contacts/contacts.component.tsx:59]: I'm not sure, so I'll ask if contact filtering should be the domain of messages? 🤔

**Review by dkarski — APPROVED** [review]
Very cool job 🍻  As part of my review, I've included a few questions that shouldn't block the pull request and shouldn't have a significant impact on the proposed changes.

File: packages/app/src/utils/sort-by-last-name.ts (hunk 2 above)
Comment (patryk-sierzega) [reply packages/app/src/utils/sort-by-last-name.ts:6]: nope :D forget about it, thanks

**Review by dkarski — APPROVED** [review]

**Pull request merged**